includes/settings/class-wcpdf-settings-debug.php

```diff
@@ -552,11 +552,31 @@ private function renumber_or_delete_document( $document, $delete_or_renumber ) {
 		if ( $document && $document->exists() ) {
 			switch ( $delete_or_renumber ) {
 				case 'renumber':
-					if ( is_callable( array( $document, 'init_number' ) ) ) {
-						$document->init_number();
-						$return = true;
-					} elseif ( 'packing-slip' === $document->get_type() && is_callable( array( WPO_WCPDF_Pro()->functions, 'init_packing_slip_number' ) ) ) {
-						WPO_WCPDF_Pro()->functions->init_packing_slip_number( $document );
+					// pro and third party documents
+					if ( 'invoice' !== $document->get_type() ) {
+						// legacy
+						if ( function_exists( 'WPO_WCPDF_Pro' ) && version_compare( WPO_WCPDF_Pro()->version, '2.15.7', '<' ) ) {
+							// packing slip
+							if ( 'packing-slip' === $document->get_type() && is_callable( array( WPO_WCPDF_Pro()->functions, 'init_packing_slip_number' ) ) ) {
+								WPO_WCPDF_Pro()->functions->init_packing_slip_number( $document );
+								$return = true;
+							// other documents
+							} elseif ( is_callable( array( $document, 'init_number' ) ) ) {
+								$document->init_number();
+								$return = true;
+							}
+						// current
```

Comment:
Can this part be removed?
I thought maybe we don't need to call the `init_packing_slip_number()` or `init_number()` while we have this update. We can call the new `add_number()` method.